spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/WriteConf.scala

```diff
@@ -52,34 +52,125 @@ case class WriteConf(batchSize: BatchSize = BatchSize.Automatic,
 
 object WriteConf {
 
-  val WriteBatchSizeInBytesProperty = "spark.cassandra.output.batch.size.bytes"
-  val WriteConsistencyLevelProperty = "spark.cassandra.output.consistency.level"
-  val WriteBatchSizeInRowsProperty = "spark.cassandra.output.batch.size.rows"
-  val WriteBatchBufferSizeProperty = "spark.cassandra.output.batch.grouping.buffer.size"
-  val WriteBatchLevelProperty = "spark.cassandra.output.batch.grouping.key"
-  val WriteParallelismLevelProperty = "spark.cassandra.output.concurrent.writes"
-  val WriteThroughputMiBPS = "spark.cassandra.output.throughput_mb_per_sec"
-  val WriteTaskMetricsProperty = "spark.cassandra.output.metrics"
+  val ReferenceSection = "Write Tuning Parameters"
 
-  // Whitelist for allowed Write environment variables
-  val Properties = Set(
+  /** Batch Size Bytes **/
+  val WriteBatchSizeInBytesProperty = "spark.cassandra.output.batch.size.bytes"
+  val DefaultBatchSizeInBytes = 1024
+  val WriteBatchSizeInBytesDescription =
+    s"""Maximum total size of the batch in bytes. Overridden by
+      |$WriteBatchSizeInRowsProperty
+    """.stripMargin
+  val BatchSizeBytesParam = ConfigParameter(
     WriteBatchSizeInBytesProperty,
+    ReferenceSection,
+    Some(DefaultBatchSizeInBytes),
+    WriteBatchSizeInBytesDescription)
+
```

Comment:
This is a nice idea with these `ConfigParameter`s. However, now that we have them, separate variables for the default value, parameter name, description are redundant and could be inlined into the config parameter object. I think this would be slightly more readable:

```scala
val BatchSizeBytesParam = ConfigParameter(
  section = ReferenceSection,
  name = "spark.cassandra.output.batch.size.bytes",
  description = "Maximum total size of the batch in bytes. Overridden by ...",
  default = 1024
)
```
(just the idea)